### ISSUE TITLE
AccessControlException running mauve

### ISSUE THREAD

**Issue by pshipton** [AdoptOpenJDK/openjdk-systemtest#284]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_special.system_ppc64le_linux_Nightly/132
MauveMultiThreadLoadTest_special_25
variation: Mode688
JVM_OPTIONS: -Xcompressedrefs -Xcompressedrefs -Xjit:count=0 -Xgcpolicy:gencon -Xaggressive -Xconcurrentlevel0 
```
01:40:52  LT  01:40:50.692 - Test failed
01:40:52  LT    Failure num.  = 1
01:40:52  LT    Test number   = 2385
01:40:52  LT    Test details  = 'Mauve[gnu.testlet.java.lang.reflect.Proxy.check13]'
01:40:52  LT    Suite number  = 0
01:40:52  LT    Thread number = 9
01:40:52  LT  >>> Captured test output >>>
01:40:52  LT  Test failed:
01:40:52  LT  java.lang.ExceptionInInitializerError
01:40:52  LT  	at java.lang.J9VMInternals.ensureError(J9VMInternals.java:148)
01:40:52  LT  	at java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:137)
01:40:52  LT  	at sun.misc.Unsafe.ensureClassInitialized(Native Method)
01:40:52  LT  	at java.lang.J9VMInternals.initialize(J9VMInternals.java:89)
01:40:52  LT  	at java.lang.Class.forName(Class.java:342)
01:40:52  LT  	at com.sun.proxy.$Proxy32.<clinit>(Unknown Source)
01:40:52  LT  	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
01:40:52  LT  	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
01:40:52  LT  	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
01:40:52  LT  	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
01:40:52  LT  	at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:739)
01:40:52  LT  	at gnu.testlet.java.lang.reflect.Proxy.check13.test(check13.java:52)
01:40:52  LT  	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
01:40:52  LT  	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
01:40:52  LT  	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
01:40:52  LT  	at java.lang.reflect.Method.invoke(Method.java:498)
01:40:52  LT  	at net.adoptopenjdk.loadTest.adaptors.MauveAdaptor.executeTest(MauveAdaptor.java:74)
01:40:52  LT  	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:182)
01:40:52  LT  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
01:40:52  LT  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
01:40:52  LT  	at java.lang.Thread.run(Thread.java:819)
01:40:52  LT  Caused by: java.security.AccessControlException: access denied: ("java.util.logging.LoggingPermission" "control")
01:40:52  LT  	at gnu.testlet.java.util.logging.Logger.TestSecurityManager.checkPermission(TestSecurityManager.java:48)
01:40:52  LT  	at java.util.logging.LogManager.checkPermission(LogManager.java:1586)
01:40:52  LT  	at java.util.logging.Logger.setParent(Logger.java:2035)
01:40:52  LT  	at java.util.logging.LogManager$7.run(LogManager.java:1205)
01:40:52  LT  	at java.security.AccessController.doPrivileged(AccessController.java:678)
01:40:52  LT  	at java.util.logging.LogManager.doSetParent(LogManager.java:1202)
01:40:52  LT  	at java.util.logging.LogManager.access$1200(LogManager.java:145)
01:40:52  LT  	at java.util.logging.LogManager$LogNode.walkAndSetParent(LogManager.java:1629)
01:40:52  LT  	at java.util.logging.LogManager$LoggerContext.addLocalLogger(LogManager.java:833)
01:40:52  LT  	at java.util.logging.LogManager$LoggerContext.ensureDefaultLogger(LogManager.java:752)
01:40:52  LT  	at java.util.logging.LogManager$LoggerContext.ensureInitialized(LogManager.java:683)
01:40:52  LT  	at java.util.logging.LogManager$LoggerContext.findLogger(LogManager.java:692)
01:40:52  LT  	at java.util.logging.LogManager$SystemLoggerContext.demandLogger(LogManager.java:922)
01:40:52  LT  	at java.util.logging.LogManager.demandSystemLogger(LogManager.java:581)
01:40:52  LT  	at java.util.logging.Logger.getPlatformLogger(Logger.java:576)
01:40:52  LT  	at java.util.logging.LoggingProxyImpl.getLogger(LoggingProxyImpl.java:41)
01:40:52  LT  	at sun.util.logging.LoggingSupport.getLogger(LoggingSupport.java:100)
01:40:52  LT  	at sun.util.logging.PlatformLogger$JavaLoggerProxy.<init>(PlatformLogger.java:602)
01:40:52  LT  	at sun.util.logging.PlatformLogger$JavaLoggerProxy.<init>(PlatformLogger.java:597)
01:40:52  LT  	at sun.util.logging.PlatformLogger.<init>(PlatformLogger.java:239)
01:40:52  LT  	at sun.util.logging.PlatformLogger.getLogger(PlatformLogger.java:198)
01:40:52  LT  	at java.awt.AWTEvent.<clinit>(AWTEvent.java:81)
01:40:52  LT  	... 19 more
```

**Comment by pshipton** [comment]
@Mesbah-Alam there must still be a test which enables the SecurityManager.

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_sanity.system_x86-64_linux_Nightly/117/tapResults/
```
LT  01:37:02.412 - Test failed
LT    Failure num.  = 1
LT    Test number   = 3607
LT    Test details  = 'Mauve[gnu.testlet.javax.security.auth.login.TestOfPR25202]'
LT    Suite number  = 0
LT    Thread number = 2
LT  >>> Captured test output >>>
LT  Test failed:
LT  java.util.ServiceConfigurationError: javax.security.auth.spi.LoginModule: Provider com.sun.jmx.remote.security.FileLoginModule could not be instantiated
LT  	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:581)
LT  	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:803)
LT  	at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:721)
LT  	at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1394)
LT  	at java.base/javax.security.auth.login.LoginContext.invoke(LoginContext.java:691)
LT  	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:665)
LT  	at java.base/javax.security.auth.login.LoginContext$4.run(LoginContext.java:663)
LT  	at java.base/java.security.AccessController.doPrivileged(AccessController.java:770)
LT  	at java.base/javax.security.auth.login.LoginContext.invokePriv(LoginContext.java:663)
LT  	at java.base/javax.security.auth.login.LoginContext.login(LoginContext.java:574)
LT  	at gnu.testlet.javax.security.auth.login.TestOfPR25202.pr25202(TestOfPR25202.java:64)
LT  	at gnu.testlet.javax.security.auth.login.TestOfPR25202.test(TestOfPR25202.java:53)
LT  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
LT  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
LT  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
LT  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
LT  	at net.adoptopenjdk.loadTest.adaptors.MauveAdaptor.executeTest(MauveAdaptor.java:74)
LT  	at net.adoptopenjdk.loadTest.LoadTestRunner$2.run(LoadTestRunner.java:182)
LT  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
LT  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
LT  	at java.base/java.lang.Thread.run(Thread.java:831)
LT  Caused by: java.lang.ExceptionInInitializerError
LT  	at java.base/java.lang.J9VMInternals.ensureError(J9VMInternals.java:193)
LT  	at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:182)
LT  	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
LT  	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
LT  	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
LT  	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
LT  	at java.base/java.util.ServiceLoader$ProviderImpl$2.run(ServiceLoader.java:787)
LT  	at java.base/java.security.AccessController.doPrivileged(AccessController.java:770)
LT  	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:792)
LT  	... 19 more
LT  Caused by: java.security.AccessControlException: access denied: ("java.util.logging.LoggingPermission" "control")
LT  	at gnu.testlet.java.util.logging.Logger.TestSecurityManager.checkPermission(TestSecurityManager.java:48)
LT  	at java.logging/java.util.logging.LogManager$LoggingProviderAccess.demandLoggerFor(LogManager.java:2707)
LT  	at java.logging/sun.util.logging.internal.LoggingProviderImpl.lambda$demandJULLoggerFor$0(LoggingProviderImpl.java:414)
LT  	at java.logging/sun.util.logging.internal.LoggingProviderImpl$$Lambda$93.0000000000000000.run(Unknown Source)
LT  	at java.base/java.security.AccessController.doPrivileged(AccessController.java:873)
LT  	at java.logging/sun.util.logging.internal.LoggingProviderImpl.demandJULLoggerFor(LoggingProviderImpl.java:415)
LT  	at java.logging/sun.util.logging.internal.LoggingProviderImpl.demandLoggerFor(LoggingProviderImpl.java:436)
LT  	at java.base/jdk.internal.logger.DefaultLoggerFinder.getLogger(DefaultLoggerFinder.java:157)
LT  	at java.base/jdk.internal.logger.LazyLoggers.lambda$getLoggerFromFinder$0(LazyLoggers.java:392)
LT  	at java.base/jdk.internal.logger.LazyLoggers$$Lambda$87.0000000000000000.run(Unknown Source)
LT  	at java.base/java.security.AccessController.doPrivileged(AccessController.java:873)
LT  	at java.base/jdk.internal.logger.LazyLoggers.getLoggerFromFinder(LazyLoggers.java:391)
LT  	at java.base/jdk.internal.logger.LazyLoggers.getLazyLogger(LazyLoggers.java:444)
LT  	at java.base/jdk.internal.logger.LazyLoggers.getLogger(LazyLoggers.java:414)
LT  	at java.base/java.lang.System.getLogger(System.java:1098)
LT  	at java.management/com.sun.jmx.remote.util.ClassLogger.<init>(ClassLogger.java:38)
LT  	at java.management/com.sun.jmx.remote.security.FileLoginModule.<clinit>(FileLoginModule.java:127)
LT  	... 26 more
```

**Comment by Mesbah-Alam** [comment]
Excluding the failing sub-tests : https://github.com/AdoptOpenJDK/openjdk-systemtest/pull/285

**Comment by pshipton** [comment]
@Mesbah-Alam the problem is not with these sub-tests. Using that approach you'll end up excluding many valid tests. The problem is with any test that enabled a SecurityManager concurrently with other running tests.

**Comment by pshipton** [comment]
@Mesbah-Alam I'm not convinced the SecurityManager is being installed by every test. Each test is called via "invoke", but in this case the running test is `gnu.testlet.java.util.logging.Logger.getAnonymousLogger.test`. I think this needs to be excluded, as well as any other test which calls `gnu.testlet.java.util.logging.Logger.TestSecurityManager.install`, and then see what other stack traces you find by using the -Xtrace option.

**Comment by Mesbah-Alam** [comment]
@pshipton , thanks. I've followed your suggestion and, it looks like, the real real culprits have been identified. Details here: https://github.com/AdoptOpenJDK/openjdk-systemtest/pull/288

**Comment by Mesbah-Alam** [comment]
https://github.com/AdoptOpenJDK/openjdk-systemtest/pull/288 has been merged. I've run a number of mauve multithreaded tests to ensure we no longer see mauve test failures due to AccessControlException for installing SecurityManeger. However, not all variations of the mauve multithreaded tests were run, and I only ran them on Linux. 

I suggest we close this issue at this point. We can re-open it if ever we see a new failure.

@pshipton
